hell/adventcalendar/2025/11/index.md

```diff
@@ -1,35 +1,119 @@
+
+```html
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:svg="http://www.w3.org/2000/svg">
+```
+
+```html
+<svg:svg viewBox="0 0 452 452">
+  <svg:title>the HTMHell logo</svg:title>
+  <svg:path d="M198.2.94c-28.6 3.3-61.4…"/>
+</svg:svg>
+```
+
+Of course, we’re talking about ePub here. There’s a [W3 ePub specification (currently at version 3.3)](https://www.w3.org/TR/epub-33/) that defines the structure and metadata of an ePub document, and gives us the [`epub:type` attribute](https://www.w3.org/TR/epub-33/#dfn-epub-type). This attribute (in conjunction with the [Structural Semantics Vocabulary specification](https://www.w3.org/TR/epub-ssv-11/)) can be used in a few ways to improve your collection’s usability within an e-reader. Let’s look at an example.
+
+One thing books often have is endnotes, but there’s no easy way of expressing that semantic state in HTML. In ePub’s vocabulary though we find the [`noteref` attribute value](https://www.w3.org/TR/epub-ssv-11/#noteref). If we use this on a link then readers know to pull in a fragment from another place and typically display then this inside a modal that can be dismissed to return to the existing place. This looks something like:
+
+```html
+<p>ePub is hellish.<a href="endnotes.xhtml#note-1" id="noteref-1" epub:type="noteref">1</a></p>
+```
+
+And in your collection of endnotes:
+
+```html
+<li id="note-1" epub:type="endnote">
+  <p>…unless you read HTMHell. <a href="page.xhtml#noteref-1" epub:type="backlink">↩</a></p>
+</li>
+```
+
+<p class="highlight"><strong>Note:</strong> <code>epub:type</code> is gradually being deprecated in favour of the <code>role</code> values defined in the <a href="https://www.w3.org/TR/dpub-aria-1.1/">Digital Publishing WAI-ARIA spec</a>, but currently that functionality is either unimplemented or only available in the latest systems. For example, the above endnotes functionality doesn’t yet work in Apple Books (generally a good e-reader) if you use <code>role="doc-noteref"</code> and <code>role="doc-endnotes"</code>.</p>
+
+As well as these ePub semantic vocabulary, we also have things like the [Z39.98-2012 Structural Semantics Vocabulary](https://www.daisy.org/z3998/2012/vocab/structure/). This extends the base ePub spec, and takes the form of `z3998`-namespaced attribute _values_, which definitely looks odd if you’re not used to it. You can use these to express even more fine-grained semantic values, for example [Roman numerals](https://www.daisy.org/z3998/2012/vocab/structure/#roman) (`<span epub:type="z3998:roman">DCLXVI</span>`) or [parts of letters](https://www.daisy.org/z3998/2012/vocab/structure/#h_letters). Broad support for extracting any functionality from these is extremely lacking, but when did that ever stop us front-end developers from going over the top?
+
+## Want to try this all out?
```

Comment:
I don't know if this is possible, but could briefly summarise the steps needed to create an epub?

1. Setup xy
2. Write XHTML
3. Use tool XY to convert